packages/gdpr-tool/src/main.js

```diff
@@ -0,0 +1,215 @@
+          const messages = await this.client.execute(request)
+
+          results = [...messages.body.results, ...results]
+        }
+        return Promise.resolve(results)
+      })
+      .catch(error => Promise.reject(error))
+  }
+
+  deleteData(customerId) {
+    if (!customerId) throw Error('missing `customerId` argument')
+    this.logger.info('Starting deletion')
+
+    const requestBuilder = createRequestBuilder({
+      projectKey: this.apiConfig.projectKey,
+    })
+    const customersUri = {
+      uri: requestBuilder.customers
+        .where(`id = "${customerId}"`)
+        .perPage(500)
+        .build(),
+      builder: requestBuilder.customers,
+    }
+    const ordersUri = {
+      uri: requestBuilder.orders
+        .where(`customerId = "${customerId}"`)
+        .perPage(500)
+        .build(),
+      builder: requestBuilder.orders,
+    }
+    const cartsUri = {
+      uri: requestBuilder.carts
+        .where(`customerId = "${customerId}"`)
+        .perPage(500)
+        .build(),
+      builder: requestBuilder.carts,
+    }
+    const paymentsUri = {
+      uri: requestBuilder.payments
+        .where(`customer(id = "${customerId}")`)
+        .perPage(500)
+        .build(),
+      builder: requestBuilder.payments,
+    }
+    const shoppingsListsUri = {
+      uri: requestBuilder.shoppingLists
+        .where(`customer(id = "${customerId}")`)
+        .perPage(500)
+        .build(),
+      builder: requestBuilder.shoppingLists,
+    }
+    const reviewsUri = {
+      uri: requestBuilder.reviews
+        .where(`customer(id = "${customerId}")`)
+        .perPage(500)
+        .build(),
+      builder: requestBuilder.reviews,
+    }
+
+    const getUris = [
+      customersUri,
+      ordersUri,
+      cartsUri,
+      paymentsUri,
+      shoppingsListsUri,
+      reviewsUri,
+    ]
+
+    // todo change to process instead of execute
+    // in case of more than 500 results
+    getUris.forEach(uri => {
+      const request = GDPRTool.buildRequest(uri.uri, 'GET')
+
+      this.client
+        .execute(request)
+        .then(async payload => {
+          if (payload.statusCode !== 200) return Promise.reject(payload)
+          GDPRTool.delete(payload, uri.builder)
+
+          return Promise.resolve()
+        })
+        .catch(error => Promise.reject(error))
+    })
+  }
+  static delete(payload, builder) {
+    const results = payload.body.results
+    if (results.length > 0) {
+      results.forEach(async result => {
+        const deleteRequest = GDPRTool.buildDeleteRequests(result, builder)
+
+        return this.client.execute(deleteRequest)
+      })
+    }
+  }
+  static buildDeleteRequests(result, builder) {
+    let deleteUri = builder
+      .byId(result.id)
+      .withVersion(result.version)
+      .build()
+
+    // todo add config option to URI builder
+    deleteUri += '&dataErasure=true'
+    return GDPRTool.buildRequest(deleteUri, 'DELETE')
```

Comment:
Why is all of this static on the prototype and not `new`ed? wouldn't it be `new GDPRTool(config)` (or whatever we call it) and then `this.buildDeletionRequest` (if renamed)`?

Reply:
Hmm, I was thinking it would be smarter to have them all on the prototype instead of duplicating(?) them all onto the instance. 

I'm not sure of the convention here so is happy to change it. Do you know how its usually structured between the two?